exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillConstExecutor.java

```diff
@@ -159,7 +164,19 @@ public void reduce(final RexBuilder rexBuilder, List<RexNode> constExps, final L
             .message(message)
             .build(logger);
         }
-        reducedValues.add(rexBuilder.makeNullLiteral(typeFactory.createSqlType(sqlTypeName)));
+
+        RelDataType type;
```

Comment:
It duplicates Drill `TypeInferenceUtils.createCalciteTypeWithNullability()`. Could you try to reuse it?